doc/dox_comments/header_files/sha256.h

```diff
@@ -244,3 +244,336 @@ int wc_Sha224Update(wc_Sha224* sha224, const byte* data, word32 len);
     \sa wc_Sha224Update
 */
 int wc_Sha224Final(wc_Sha224* sha224, byte* hash);
+
+/*!
+    \ingroup SHA
+    \brief Initializes SHA256 with heap and device ID.
+
+    \return 0 on success
+    \return negative on error
+
+    \param sha SHA256 structure
+    \param heap Heap hint
+    \param devId Device ID
+
+    _Example_
+    \code
+    wc_Sha256 sha;
+    int ret = wc_InitSha256_ex(&sha, NULL, INVALID_DEVID);
+    \endcode
+
+    \sa wc_InitSha256
+*/
+int wc_InitSha256_ex(wc_Sha256* sha, void* heap, int devId);
+
+/*!
+    \ingroup SHA
+    \brief Gets raw hash without finalizing.
+
+    \return 0 on success
+    \return negative on error
+
+    \param sha256 SHA256 structure
+    \param hash Output hash buffer
+
+    _Example_
+    \code
+    wc_Sha256 sha;
+    byte hash[WC_SHA256_DIGEST_SIZE];
+    int ret = wc_Sha256FinalRaw(&sha, hash);
+    \endcode
+
+    \sa wc_Sha256Final
+*/
+int wc_Sha256FinalRaw(wc_Sha256* sha256, byte* hash);
+
+/*!
+    \ingroup SHA
+    \brief Transforms SHA256 block.
+
+    \return 0 on success
+    \return negative on error
+
+    \param sha SHA256 structure
+    \param data Block data
+
+    _Example_
+    \code
+    wc_Sha256 sha;
+    unsigned char block[WC_SHA256_BLOCK_SIZE];
+    int ret = wc_Sha256Transform(&sha, block);
+    \endcode
+
+    \sa wc_Sha256Update
+*/
+int wc_Sha256Transform(wc_Sha256* sha, const unsigned char* data);
+
+/*!
+    \ingroup SHA
+    \brief Hashes single block and outputs result.
+
+    \return 0 on success
+    \return negative on error
+
+    \param sha SHA256 structure
+    \param data Block data
+    \param hash Output hash buffer
+
+    _Example_
+    \code
+    wc_Sha256 sha;
+    unsigned char block[WC_SHA256_BLOCK_SIZE];
+    unsigned char hash[WC_SHA256_DIGEST_SIZE];
+    int ret = wc_Sha256HashBlock(&sha, block, hash);
+    \endcode
+
+    \sa wc_Sha256Transform
+*/
+int wc_Sha256HashBlock(wc_Sha256* sha, const unsigned char* data,
+    unsigned char* hash);
+
+/*!
+    \ingroup SHA
+    \brief Grows SHA256 buffer with input data.
```

Comment:
There is very specific build conditions for wc_Sha256Grow. They should be listed in the brief along with the purpose of the function. It is used more for when keeping an internal buffer to hold all data to be hashed rather than itterating over update due to some hardware acceleration restrictions.